### PR TITLE
Variable scale and reference

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Adds `scale` and `reference` attributes to a variable, which are used to multiply and/or add a value to a variable to make the resulting numerics better conditioned.
Progress towards #2418 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/discretisations/discretisation.py [hunk 1]
```diff
@@ -739,6 +741,9 @@ def process_dict(self, var_eqn_dict):
             Equations ({variable: equation} dict) to dicretise
             (can be model.rhs, model.algebraic, model.initial_conditions or
             model.variables)
+        ics : bool, optional
+            Whether the equations are initial conditions. If True, the equations are
```
Comment (brosaplanella) [inline pybamm/discretisations/discretisation.py:745]:
```suggestion
            Whether the equations have initial conditions. If True, the equations are
```

File: pybamm/parameters/lead_acid_parameters.py [hunk 2]
```diff
@@ -422,7 +422,7 @@ def kappa_e(self, c_e, T):
         kappa_scale = self.F**2 * self.D_e_typ * self.c_e_typ / (self.R * self.T_ref)
         return self.kappa_e_dimensional(c_e_dimensional, self.T_ref) / kappa_scale
 
-    def chiT_over_c(self, c_e, T):
+    def chiRT_over_Fc(self, c_e, T):
```
Comment (brosaplanella) [inline pybamm/parameters/lead_acid_parameters.py:425]: What's the meaning of Fc?

**Review by brosaplanella — APPROVED** [review]
Looks good, but the dimensional DFN is not there yet, is it?

File: pybamm/parameters/lead_acid_parameters.py (hunk 2 above)
Comment (tinosulzer) [reply pybamm/parameters/lead_acid_parameters.py:425]: Faraday's constant times c

**Comment by tinosulzer** [comment]
I removed the dimensional DFN since when we eventually dimensionalize all the models it will not be there, so I don't want people adding it to their code. This PR now only adds the scaling and reference

File: pybamm/discretisations/discretisation.py (hunk 1 above)
Comment (tinosulzer) [reply pybamm/discretisations/discretisation.py:745]: This is meant to be `are`, this is the function that discretises the equations (rhs, algebraic, initial conditions) and there is a special case for initial conditions (reference only gets taken away from initial conditions)

**Pull request merged**